src/EventStore.Common/Configuration/OptionsDumper.cs

```diff
@@ -8,15 +9,14 @@
 namespace EventStore.Common.Configuration {
 	public class OptionsDumper {
 		private readonly IEnumerable<Type> _optionSections;
-		private readonly List<string> _sortOrder;
-
+		private readonly List<IGrouping<string, PropertyInfo>> _sortGroupBy;
+		
 		public OptionsDumper(IEnumerable<Type> optionSections) {
 			_optionSections = optionSections;
-			_sortOrder = _optionSections.SelectMany(optionSection =>
-					optionSection.GetProperties()
-						.Select(p => NameTranslators.CombineByPascalCase(p.Name, " ").ToUpper()))
-				.OrderBy(x => x)
-				.ToList();
+			_sortGroupBy = _optionSections.SelectMany(type => type.GetProperties().OrderBy(x => x.Name).GroupBy(
+					option =>
+						option.DeclaringType?.GetCustomAttribute<DescriptionAttribute>()?.Description ?? string.Empty))
+				.OrderBy(option => option.Key).ToList();
```

Comment:
getting the declaring type for each property is feeling a bit odd because we already have it (it is `type`) because we are, indeed, iterating through its properties

it might be more natural to to map each optionSection to a tuple containing its description and the names, in order, of the properties.

then _sortGroupBy would be of type `List<(string, List<string>)>` (and might have a different name), and you wouldn't need the `SelectMany` or the `GroupBy`

I _think_ you can call `CombineByPascalCase` in here like before, instead of having to call `RemoveWhiteSpacesToCombine` later